p5/core/primitives.py

```diff
@@ -477,19 +498,24 @@ def square(*args, mode=None):
     :raises ValueError: When the mode is set to 'CORNERS'
 
     """
-    if len(args) == 2:
-        coordinate, side_length = args
-    elif len(args) == 3:
-        coordinate, side_length = args[:2], args[2]
-    else:
-        raise ValueError("Unexpected number of arguments passed to square()")
+    if builtins.current_renderer == 'vispy':
+        if len(args) == 2:
+            coordinate, side_length = args
+        elif len(args) == 3:
+            coordinate, side_length = args[:2], args[2]
+        else:
+            raise ValueError("Unexpected number of arguments passed to square()")
 
-    if mode is None:
-        mode = _rect_mode
+        if mode is None:
+            mode = p5.renderer.style.rect_mode
+
+        if mode == 'CORNERS':
+            raise ValueError("Cannot draw square with {} mode".format(mode))
+        return rect(coordinate, side_length, side_length, mode=mode)
+    elif builtins.current_renderer == 'skia':
+        if should_draw():
+            rect(*args)
```

Comment:
I think `rect` expects there to be at least 4 input parameters, whereas `square` has only 2 or 3?
https://p5.readthedocs.io/en/latest/reference/shape.html?#square

Reply:
Yes, fixed. Thanks